content/en-US/api/roundRect.md

```diff
@@ -0,0 +1,113 @@
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
+  vertically along `x`, that means the radius values that applied to the top corners are now applied
+  to the bottom.</dd>
+
+  <dt><em>radii</em></dt>
+  <dd>Each value `r` in `radii`  could be a number or an object with `{x, y}` properties. If `r` is 
+  a number, the corresponding corner(s) are drawn as a cicurlar arc with radius `r`; if `r` is an
+  object, the correspondinf corner(s) are drawn as 
+  [elliptical arc](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/ellipse)
+  whose `radiusX` and `radiusY` are equal to `x` and `y`, respectively. `r`, `x` or `y` must be
+  non-negative:
+
+  * If `radii`'s size is 1, then arc created by `radii[0]` replaces all 4 corners of rectangle. 
+
+  * If `radii`'s size is 2, then the arc created by `radii[0]` replaces the upper left and lower
+  right corners of the rectangle; the arc created by `radii[1]` replaces the upper right and lower
+  left corners of rectangle.
+
+  * If `radii`'s size is 3, then the arc created by `radii[0]` replaces the upper left corner; the
+  arc created by `radii[1]` replaces the upper right and lower left corners; and the arc created by
+  `radii[2]` replaces the lower right corner of rectangle.
+
+
+  * If `radii`'s size is 4, then the arc createed by radii[0], radii[1], radii[2] and radii[3]
+  replaces upper left, upper right, lower right and lower left corner of rectangle, respectively.
+
+  * If `radii`'s size is not any of the listed values above, then roundRect aborts and returns an
+  RangeError.
+
+  Note that if the sum of the radii of two corners of the same edge is greater than the length of
+  that edge, all the `radii` of the rounded rectangle are scaled by a factor of length / (r1 + r2).
+</dd>
+</dl>
+
+## Examples
+
+
+### Drawing roundRect with circular arc
+
+The following example draws 4 rooundRect with `raddi`'s size equals to 1, 2, 3 or 4 and `r` is in
+form of double.
+
+```js
+const canvas = document.createElement('canvas');
+canvas.width = 1000;
+canvas.height = 500;
+const ctx = canvas.getContext('2d');
+document.body.appendChild(canvas);
+ctx.strokeStyle = '#0f0';
+ctx.lineWidth = 5;
+ctx.roundRect(50, 50, 100, 100, [20]);
+ctx.roundRect(200, 50, 100, 100, [20, 40]);
+ctx.roundRect(50, 200, 100, 100, [10, 25, 40]);
+ctx.roundRect(200, 200, 100, 100, [5, 15, 30, 50]);
+ctx.stroke();
+```
+
+### Drawing roundRect with elliptical arc
+
+The following example draws rooundRect with different elliptical arc.
+
+```js
+const canvas = document.createElement('canvas');
+canvas.width = 500;
+canvas.height = 500;
+const ctx = canvas.getContext('2d');
+document.body.appendChild(canvas);
+ctx.strokeStyle = '#0f0';
+ctx.lineWidth = 5;
+ctx.roundRect(50, 50, 120, 120, [new DOMPoint(25, 50)]);
+ctx.roundRect(210, 50, 120, 120, [new DOMPoint(20, 50), new DOMPoint(50, 20), new DOMPoint(20, 50), 
+  new DOMPoint(50, 20)]);
```

Comment:
Wanna do some examples with `DOMPointInit`? Also some examples mixing doubles and `DOMPointInit`s e.g.:

`ctx.roundRect(10, 10, 100, 100, [{25, 50}, 10])`

Reply:
good idea!